### ISSUE TITLE
EfficientNet official VS non-official implementation + Reproducibility 

### ISSUE THREAD

**Issue by innat** [tensorflow/tensorflow#48103]
I've encountered weird phenomena with the official implementation of the `EfficientNet` model. The **validation score** doesn't improve as expected and far less than a non-official one. 

# Reproduce Issue

```
# Install non-official efficient network 
!pip install -U git+https://github.com/qubvel/efficientnet
```

```
import os
import numpy as np
import tensorflow as tf
print(tf.__version__)
import efficientnet.keras as efn 

batch_size = 32 
num_classes = 10
epochs = 3
official_efficient_net = False
```

```
## Model 
input_shape = (32,32,3)
if official_efficient_net:
    base_model  = tf.keras.applications.efficientnet.EfficientNetB0(include_top=False,
                                                                    weights="imagenet", 
                                                                    input_shape=input_shape)
else:
    base_model = efn.EfficientNetB0(include_top=False,
                                        weights="imagenet", 
                                        input_shape=input_shape)

global_average_layer = tf.keras.layers.GlobalAveragePooling2D()
dense_layer = tf.keras.layers.Dense(10, activation='softmax')
Model  = tf.keras.Sequential([base_model, global_average_layer, dense_layer])
```

```
## Training Data
(x_train, y_train), (x_test, y_test) = keras.datasets.cifar10.load_data()
y_train = tf.keras.utils.to_categorical(y_train, num_classes)
y_test = tf.keras.utils.to_categorical(y_test, num_classes)
x_train = x_train.astype('float32') / 255
x_test = x_test.astype('float32') / 255

## Training
Model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])
cnn = Model.fit(x_train, y_train, batch_size=batch_size, 
                epochs=epochs, validation_data=(x_test,y_test))
```

Tested in `TensorFlow 2.4.1`. 
Training log from **Official EfficientNet**
>
Epoch 1/3
1563/1563 [==============================] - 128s 77ms/step - loss: 1.6419 - accuracy: 0.4368 - val_loss: 2.4821 - val_accuracy: 0.1817
Epoch 2/3
1563/1563 [==============================] - 119s 76ms/step - loss: 0.9287 - accuracy: 0.6800 - val_loss: 3.6128 - val_accuracy: 0.1184
Epoch 3/3
1563/1563 [==============================] - 119s 76ms/step - loss: 0.7524 - accuracy: 0.7478 - val_loss: 3.9361 - val_accuracy: 0.1000

Training log from **Non-official EfficientNet**
>
Epoch 1/3
1563/1563 [==============================] - 128s 77ms/step - loss: 1.3312 - accuracy: 0.5551 - val_loss: 0.7306 - val_accuracy: 0.7449
Epoch 2/3
1563/1563 [==============================] - 119s 76ms/step - loss: 0.7034 - accuracy: 0.7614 - val_loss: 0.6250 - val_accuracy: 0.7846
Epoch 3/3
1563/1563 [==============================] - 119s 76ms/step - loss: 0.5726 - accuracy: 0.8098 - val_loss: 0.6379 - val_accuracy: 0.7813

I've also tested with other ImageNet models (ie. `RestNet`, `DenseNet`). They all give a reasonable validation score. 

# Reproducibility 

Apart from the above issue, there is another issue about reproducibility from `EfficientNet`. It was asked before, here [Issue #47174](https://github.com/tensorflow/tensorflow/issues/47174). I've tested with `TF 2.4.1` and I also face the same problem.

**Comment by fsx950223** [comment]
Remove
```
x_train = x_train.astype('float32') / 255
x_test = x_test.astype('float32') / 255
```
fix the bug.

**Comment by innat** [comment]
Sorry, couldn't catch it. Isn't it normalizing the data? Are you saying to remove it?

**Comment by Saduf2019** [comment]
@innat 
When we try to replicate the code shared i face a different error, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/eeba421a301ad30bfe87b5291059b5b6/untitled580.ipynb). If possible share a colab gist with the issue reported.

**Comment by innat101** [comment]
@Saduf2019, as I mentioned in my post. You forgot to install a non-official efficient net. Add this at the beginning. 

```
!pip install -U git+https://github.com/qubvel/efficientnet
```

**Comment by innat101** [comment]
If you encounter any issue with this non-official package, then please see the read me at least of it, here https://github.com/qubvel/efficientnet

It's important in case you choose `tf` version (maybe).

**Comment by innat** [comment]
@Saduf2019 

Install it at the beginning 

```
# Install non-official efficient network 
!pip install -U git+https://github.com/qubvel/efficientnet
```

If you encounter any issue, then please check the read me of this non-official efnet.  https://github.com/qubvel/efficientnet#examples
or, https://github.com/qubvel/efficientnet#important

**Comment by innat** [comment]
@fsx950223 

After removing the 

```
x_train = x_train.astype('float32') / 255
x_test = x_test.astype('float32') / 255
```

the problem is gone. Now, the official implementation of an efficient net gives reasonable validation results. Will it be fixed soon?

**Comment by Saduf2019** [comment]
I am able to run the code on tf 2.4 but throws error on nightly, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/1dcbc895897a50056f5882493622cf2d/untitled582.ipynb).
Thanks!

**Comment by innat** [comment]
@Saduf2019 sorry for the inconvenience. I was about to share working colab but in the meantime, I also tried one thing and this time everything was OK. The things are what @fsx950223 suggested to remove the following operation for the official efficientnet.

```
x_train = x_train.astype('float32') / 255
x_test = x_test.astype('float32') / 255
``

When I add the above normalization, the efficientnet gave wearied results on the validation set. But when I didn't use this normalization, the efficientnet gave reasonable results on the validation set. So, now I am wondering, is it some kind of bug?
